src/client/QXmppTransferManager.cpp

```diff
@@ -1342,8 +1342,8 @@ void QXmppTransferManager::streamInitiationResultReceived(const QXmppStreamIniti
         job->state() != QXmppTransferJob::OfferState)
         return;
 
-    const auto &form = iq.featureForm();
-    const auto &fields = form.fields();
+    const auto form = iq.featureForm();
+    const auto fields = form.fields();
```

Comment:
I'm not sure why this is dangerous. `featureForm()` returns by value, so `const auto &form` would extend the lifetime, and so would `const auto &fields` for the return value of `fields()` (which would be by-value, since the `const` overload is chosen).

Reply:
To be honest I'm not completely sure anymore if this is a problem, maybe you know more on this than I do.
If the lifetime is extended by taking the ref, it should be fine I think.